// @bnb-chain/greenfield-cosmos-types/cosmos/slashing/v1beta1/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/cosmos.slashing.v1beta1.MsgUnjail", tx_1.MsgUnjail], ["/cosmos.slashing.v1beta1.MsgUpdateParams", tx_1.MsgUpdateParams], ["/cosmos.slashing.v1beta1.MsgImpeach", tx_1.MsgImpeach]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        unjail(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUnjail",
                value: tx_1.MsgUnjail.encode(value).finish()
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        },
        impeach(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgImpeach",
                value: tx_1.MsgImpeach.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        unjail(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUnjail",
                value
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUpdateParams",
                value
            };
        },
        impeach(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgImpeach",
                value
            };
        }
    },
    toJSON: {
        unjail(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUnjail",
                value: tx_1.MsgUnjail.toJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        },
        impeach(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgImpeach",
                value: tx_1.MsgImpeach.toJSON(value)
            };
        }
    },
    fromJSON: {
        unjail(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUnjail",
                value: tx_1.MsgUnjail.fromJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        },
        impeach(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgImpeach",
                value: tx_1.MsgImpeach.fromJSON(value)
            };
        }
    },
    fromPartial: {
        unjail(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUnjail",
                value: tx_1.MsgUnjail.fromPartial(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        },
        impeach(value) {
            return {
                typeUrl: "/cosmos.slashing.v1beta1.MsgImpeach",
                value: tx_1.MsgImpeach.fromPartial(value)
            };
        }
    }
};
