// @bnb-chain/greenfield-cosmos-types/cosmos/oracle/v1/query.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryClientImpl = exports.QueryInturnRelayerResponse = exports.QueryInturnRelayerRequest = exports.QueryParamsResponse = exports.QueryParamsRequest = exports.claimSrcChainToJSON = exports.claimSrcChainFromJSON = exports.ClaimSrcChainAmino = exports.ClaimSrcChainSDKType = exports.ClaimSrcChain = exports.protobufPackage = void 0;
//@ts-nocheck
/* eslint-disable */
const oracle_1 = require("./oracle");
const _m0 = __importStar(require("protobufjs/minimal"));
const helpers_1 = require("../../../helpers");
exports.protobufPackage = "cosmos.oracle.v1";
/** ClaimSrcChain defines the src chain of a claim */
var ClaimSrcChain;
(function (ClaimSrcChain) {
    /** CLAIM_SRC_CHAIN_UNSPECIFIED - CLAIM_SRC_CHAIN_UNSPECIFIED */
    ClaimSrcChain[ClaimSrcChain["CLAIM_SRC_CHAIN_UNSPECIFIED"] = 0] = "CLAIM_SRC_CHAIN_UNSPECIFIED";
    /** CLAIM_SRC_CHAIN_BSC - CLAIM_SRC_CHAIN_BSC defines BSC source chain */
    ClaimSrcChain[ClaimSrcChain["CLAIM_SRC_CHAIN_BSC"] = 1] = "CLAIM_SRC_CHAIN_BSC";
    /** CLAIM_SRC_CHAIN_OP_BNB - CLAIM_SRC_CHAIN_OP_BNB defines OPBNB source chain */
    ClaimSrcChain[ClaimSrcChain["CLAIM_SRC_CHAIN_OP_BNB"] = 2] = "CLAIM_SRC_CHAIN_OP_BNB";
    ClaimSrcChain[ClaimSrcChain["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ClaimSrcChain = exports.ClaimSrcChain || (exports.ClaimSrcChain = {}));
exports.ClaimSrcChainSDKType = ClaimSrcChain;
exports.ClaimSrcChainAmino = ClaimSrcChain;
function claimSrcChainFromJSON(object) {
    switch (object) {
        case 0:
        case "CLAIM_SRC_CHAIN_UNSPECIFIED":
            return ClaimSrcChain.CLAIM_SRC_CHAIN_UNSPECIFIED;
        case 1:
        case "CLAIM_SRC_CHAIN_BSC":
            return ClaimSrcChain.CLAIM_SRC_CHAIN_BSC;
        case 2:
        case "CLAIM_SRC_CHAIN_OP_BNB":
            return ClaimSrcChain.CLAIM_SRC_CHAIN_OP_BNB;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ClaimSrcChain.UNRECOGNIZED;
    }
}
exports.claimSrcChainFromJSON = claimSrcChainFromJSON;
function claimSrcChainToJSON(object) {
    switch (object) {
        case ClaimSrcChain.CLAIM_SRC_CHAIN_UNSPECIFIED:
            return "CLAIM_SRC_CHAIN_UNSPECIFIED";
        case ClaimSrcChain.CLAIM_SRC_CHAIN_BSC:
            return "CLAIM_SRC_CHAIN_BSC";
        case ClaimSrcChain.CLAIM_SRC_CHAIN_OP_BNB:
            return "CLAIM_SRC_CHAIN_OP_BNB";
        case ClaimSrcChain.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.claimSrcChainToJSON = claimSrcChainToJSON;
function createBaseQueryParamsRequest() {
    return {};
}
exports.QueryParamsRequest = {
    typeUrl: "/cosmos.oracle.v1.QueryParamsRequest",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryParamsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    fromPartial(_) {
        const message = createBaseQueryParamsRequest();
        return message;
    },
    fromSDK(_) {
        return {};
    },
    toSDK(_) {
        const obj = {};
        return obj;
    },
    fromAmino(_) {
        const message = createBaseQueryParamsRequest();
        return message;
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.QueryParamsRequest.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/QueryParamsRequest",
            value: exports.QueryParamsRequest.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.QueryParamsRequest.decode(message.value);
    },
    toProto(message) {
        return exports.QueryParamsRequest.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.oracle.v1.QueryParamsRequest",
            value: exports.QueryParamsRequest.encode(message).finish()
        };
    }
};
function createBaseQueryParamsResponse() {
    return {
        params: oracle_1.Params.fromPartial({})
    };
}
exports.QueryParamsResponse = {
    typeUrl: "/cosmos.oracle.v1.QueryParamsResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.params !== undefined) {
            oracle_1.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.params = oracle_1.Params.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: (0, helpers_1.isSet)(object.params) ? oracle_1.Params.fromJSON(object.params) : undefined
        };
    },
    toJSON(message) {
        const obj = {};
        message.params !== undefined && (obj.params = message.params ? oracle_1.Params.toJSON(message.params) : undefined);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseQueryParamsResponse();
        message.params = object.params !== undefined && object.params !== null ? oracle_1.Params.fromPartial(object.params) : undefined;
        return message;
    },
    fromSDK(object) {
        return {
            params: object.params ? oracle_1.Params.fromSDK(object.params) : undefined
        };
    },
    toSDK(message) {
        const obj = {};
        message.params !== undefined && (obj.params = message.params ? oracle_1.Params.toSDK(message.params) : undefined);
        return obj;
    },
    fromAmino(object) {
        const message = createBaseQueryParamsResponse();
        if (object.params !== undefined && object.params !== null) {
            message.params = oracle_1.Params.fromAmino(object.params);
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.params = message.params ? oracle_1.Params.toAmino(message.params) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.QueryParamsResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/QueryParamsResponse",
            value: exports.QueryParamsResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.QueryParamsResponse.decode(message.value);
    },
    toProto(message) {
        return exports.QueryParamsResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.oracle.v1.QueryParamsResponse",
            value: exports.QueryParamsResponse.encode(message).finish()
        };
    }
};
function createBaseQueryInturnRelayerRequest() {
    return {
        claimSrcChain: 0
    };
}
exports.QueryInturnRelayerRequest = {
    typeUrl: "/cosmos.oracle.v1.QueryInturnRelayerRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.claimSrcChain !== 0) {
            writer.uint32(8).int32(message.claimSrcChain);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryInturnRelayerRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.claimSrcChain = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            claimSrcChain: (0, helpers_1.isSet)(object.claimSrcChain) ? claimSrcChainFromJSON(object.claimSrcChain) : -1
        };
    },
    toJSON(message) {
        const obj = {};
        message.claimSrcChain !== undefined && (obj.claimSrcChain = claimSrcChainToJSON(message.claimSrcChain));
        return obj;
    },
    fromPartial(object) {
        const message = createBaseQueryInturnRelayerRequest();
        message.claimSrcChain = object.claimSrcChain ?? 0;
        return message;
    },
    fromSDK(object) {
        return {
            claimSrcChain: (0, helpers_1.isSet)(object.claim_src_chain) ? claimSrcChainFromJSON(object.claim_src_chain) : -1
        };
    },
    toSDK(message) {
        const obj = {};
        message.claimSrcChain !== undefined && (obj.claim_src_chain = claimSrcChainToJSON(message.claimSrcChain));
        return obj;
    },
    fromAmino(object) {
        const message = createBaseQueryInturnRelayerRequest();
        if (object.claim_src_chain !== undefined && object.claim_src_chain !== null) {
            message.claimSrcChain = claimSrcChainFromJSON(object.claim_src_chain);
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.claim_src_chain = claimSrcChainToJSON(message.claimSrcChain);
        return obj;
    },
    fromAminoMsg(object) {
        return exports.QueryInturnRelayerRequest.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/QueryInturnRelayerRequest",
            value: exports.QueryInturnRelayerRequest.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.QueryInturnRelayerRequest.decode(message.value);
    },
    toProto(message) {
        return exports.QueryInturnRelayerRequest.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.oracle.v1.QueryInturnRelayerRequest",
            value: exports.QueryInturnRelayerRequest.encode(message).finish()
        };
    }
};
function createBaseQueryInturnRelayerResponse() {
    return {
        blsPubKey: "",
        relayInterval: undefined
    };
}
exports.QueryInturnRelayerResponse = {
    typeUrl: "/cosmos.oracle.v1.QueryInturnRelayerResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.blsPubKey !== "") {
            writer.uint32(10).string(message.blsPubKey);
        }
        if (message.relayInterval !== undefined) {
            oracle_1.RelayInterval.encode(message.relayInterval, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryInturnRelayerResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.blsPubKey = reader.string();
                    break;
                case 2:
                    message.relayInterval = oracle_1.RelayInterval.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            blsPubKey: (0, helpers_1.isSet)(object.blsPubKey) ? String(object.blsPubKey) : "",
            relayInterval: (0, helpers_1.isSet)(object.relayInterval) ? oracle_1.RelayInterval.fromJSON(object.relayInterval) : undefined
        };
    },
    toJSON(message) {
        const obj = {};
        message.blsPubKey !== undefined && (obj.blsPubKey = message.blsPubKey);
        message.relayInterval !== undefined && (obj.relayInterval = message.relayInterval ? oracle_1.RelayInterval.toJSON(message.relayInterval) : undefined);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseQueryInturnRelayerResponse();
        message.blsPubKey = object.blsPubKey ?? "";
        message.relayInterval = object.relayInterval !== undefined && object.relayInterval !== null ? oracle_1.RelayInterval.fromPartial(object.relayInterval) : undefined;
        return message;
    },
    fromSDK(object) {
        return {
            blsPubKey: object?.bls_pub_key,
            relayInterval: object.relay_interval ? oracle_1.RelayInterval.fromSDK(object.relay_interval) : undefined
        };
    },
    toSDK(message) {
        const obj = {};
        obj.bls_pub_key = message.blsPubKey;
        message.relayInterval !== undefined && (obj.relay_interval = message.relayInterval ? oracle_1.RelayInterval.toSDK(message.relayInterval) : undefined);
        return obj;
    },
    fromAmino(object) {
        const message = createBaseQueryInturnRelayerResponse();
        if (object.bls_pub_key !== undefined && object.bls_pub_key !== null) {
            message.blsPubKey = object.bls_pub_key;
        }
        if (object.relay_interval !== undefined && object.relay_interval !== null) {
            message.relayInterval = oracle_1.RelayInterval.fromAmino(object.relay_interval);
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.bls_pub_key = message.blsPubKey;
        obj.relay_interval = message.relayInterval ? oracle_1.RelayInterval.toAmino(message.relayInterval) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.QueryInturnRelayerResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/QueryInturnRelayerResponse",
            value: exports.QueryInturnRelayerResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.QueryInturnRelayerResponse.decode(message.value);
    },
    toProto(message) {
        return exports.QueryInturnRelayerResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.oracle.v1.QueryInturnRelayerResponse",
            value: exports.QueryInturnRelayerResponse.encode(message).finish()
        };
    }
};
class QueryClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.Params = this.Params.bind(this);
        this.InturnRelayer = this.InturnRelayer.bind(this);
    }
    Params(request = {}) {
        const data = exports.QueryParamsRequest.encode(request).finish();
        const promise = this.rpc.request("cosmos.oracle.v1.Query", "Params", data);
        return promise.then(data => exports.QueryParamsResponse.decode(new _m0.Reader(data)));
    }
    InturnRelayer(request) {
        const data = exports.QueryInturnRelayerRequest.encode(request).finish();
        const promise = this.rpc.request("cosmos.oracle.v1.Query", "InturnRelayer", data);
        return promise.then(data => exports.QueryInturnRelayerResponse.decode(new _m0.Reader(data)));
    }
}
exports.QueryClientImpl = QueryClientImpl;
