// @bnb-chain/greenfield-cosmos-types/cosmos/oracle/v1/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/cosmos.oracle.v1.MsgClaim", tx_1.MsgClaim], ["/cosmos.oracle.v1.MsgUpdateParams", tx_1.MsgUpdateParams]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        claim(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgClaim",
                value: tx_1.MsgClaim.encode(value).finish()
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        claim(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgClaim",
                value
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgUpdateParams",
                value
            };
        }
    },
    toJSON: {
        claim(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgClaim",
                value: tx_1.MsgClaim.toJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        }
    },
    fromJSON: {
        claim(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgClaim",
                value: tx_1.MsgClaim.fromJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        }
    },
    fromPartial: {
        claim(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgClaim",
                value: tx_1.MsgClaim.fromPartial(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/cosmos.oracle.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        }
    }
};
