// @bnb-chain/greenfield-cosmos-types/cosmos/crosschain/v1/tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgClientImpl = exports.MsgMintModuleTokensResponse = exports.MsgMintModuleTokens = exports.MsgUpdateChannelPermissionsResponse = exports.MsgUpdateChannelPermissions = exports.MsgUpdateParamsResponse = exports.MsgUpdateParams = exports.protobufPackage = void 0;
//@ts-nocheck
/* eslint-disable */
const crosschain_1 = require("./crosschain");
const _m0 = __importStar(require("protobufjs/minimal"));
const helpers_1 = require("../../../helpers");
exports.protobufPackage = "cosmos.crosschain.v1";
function createBaseMsgUpdateParams() {
    return {
        authority: "",
        params: crosschain_1.Params.fromPartial({})
    };
}
exports.MsgUpdateParams = {
    typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            crosschain_1.Params.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.params = crosschain_1.Params.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: (0, helpers_1.isSet)(object.authority) ? String(object.authority) : "",
            params: (0, helpers_1.isSet)(object.params) ? crosschain_1.Params.fromJSON(object.params) : undefined
        };
    },
    toJSON(message) {
        const obj = {};
        message.authority !== undefined && (obj.authority = message.authority);
        message.params !== undefined && (obj.params = message.params ? crosschain_1.Params.toJSON(message.params) : undefined);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseMsgUpdateParams();
        message.authority = object.authority ?? "";
        message.params = object.params !== undefined && object.params !== null ? crosschain_1.Params.fromPartial(object.params) : undefined;
        return message;
    },
    fromSDK(object) {
        return {
            authority: object?.authority,
            params: object.params ? crosschain_1.Params.fromSDK(object.params) : undefined
        };
    },
    toSDK(message) {
        const obj = {};
        obj.authority = message.authority;
        message.params !== undefined && (obj.params = message.params ? crosschain_1.Params.toSDK(message.params) : undefined);
        return obj;
    },
    fromAmino(object) {
        const message = createBaseMsgUpdateParams();
        if (object.authority !== undefined && object.authority !== null) {
            message.authority = object.authority;
        }
        if (object.params !== undefined && object.params !== null) {
            message.params = crosschain_1.Params.fromAmino(object.params);
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.authority = message.authority;
        obj.params = message.params ? crosschain_1.Params.toAmino(message.params) : crosschain_1.Params.fromPartial({});
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgUpdateParams.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/MsgUpdateParams",
            value: exports.MsgUpdateParams.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgUpdateParams.decode(message.value);
    },
    toProto(message) {
        return exports.MsgUpdateParams.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
            value: exports.MsgUpdateParams.encode(message).finish()
        };
    }
};
function createBaseMsgUpdateParamsResponse() {
    return {};
}
exports.MsgUpdateParamsResponse = {
    typeUrl: "/cosmos.crosschain.v1.MsgUpdateParamsResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateParamsResponse();
        return message;
    },
    fromSDK(_) {
        return {};
    },
    toSDK(_) {
        const obj = {};
        return obj;
    },
    fromAmino(_) {
        const message = createBaseMsgUpdateParamsResponse();
        return message;
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgUpdateParamsResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/MsgUpdateParamsResponse",
            value: exports.MsgUpdateParamsResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgUpdateParamsResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgUpdateParamsResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.crosschain.v1.MsgUpdateParamsResponse",
            value: exports.MsgUpdateParamsResponse.encode(message).finish()
        };
    }
};
function createBaseMsgUpdateChannelPermissions() {
    return {
        authority: "",
        channelPermissions: []
    };
}
exports.MsgUpdateChannelPermissions = {
    typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        for (const v of message.channelPermissions) {
            crosschain_1.ChannelPermission.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateChannelPermissions();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.channelPermissions.push(crosschain_1.ChannelPermission.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: (0, helpers_1.isSet)(object.authority) ? String(object.authority) : "",
            channelPermissions: Array.isArray(object?.channelPermissions) ? object.channelPermissions.map((e) => crosschain_1.ChannelPermission.fromJSON(e)) : []
        };
    },
    toJSON(message) {
        const obj = {};
        message.authority !== undefined && (obj.authority = message.authority);
        if (message.channelPermissions) {
            obj.channelPermissions = message.channelPermissions.map(e => e ? crosschain_1.ChannelPermission.toJSON(e) : undefined);
        }
        else {
            obj.channelPermissions = [];
        }
        return obj;
    },
    fromPartial(object) {
        const message = createBaseMsgUpdateChannelPermissions();
        message.authority = object.authority ?? "";
        message.channelPermissions = object.channelPermissions?.map(e => crosschain_1.ChannelPermission.fromPartial(e)) || [];
        return message;
    },
    fromSDK(object) {
        return {
            authority: object?.authority,
            channelPermissions: Array.isArray(object?.channel_permissions) ? object.channel_permissions.map((e) => crosschain_1.ChannelPermission.fromSDK(e)) : []
        };
    },
    toSDK(message) {
        const obj = {};
        obj.authority = message.authority;
        if (message.channelPermissions) {
            obj.channel_permissions = message.channelPermissions.map(e => e ? crosschain_1.ChannelPermission.toSDK(e) : undefined);
        }
        else {
            obj.channel_permissions = [];
        }
        return obj;
    },
    fromAmino(object) {
        const message = createBaseMsgUpdateChannelPermissions();
        if (object.authority !== undefined && object.authority !== null) {
            message.authority = object.authority;
        }
        message.channelPermissions = object.channel_permissions?.map(e => crosschain_1.ChannelPermission.fromAmino(e)) || [];
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.authority = message.authority;
        if (message.channelPermissions) {
            obj.channel_permissions = message.channelPermissions.map(e => e ? crosschain_1.ChannelPermission.toAmino(e) : undefined);
        }
        else {
            obj.channel_permissions = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgUpdateChannelPermissions.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/MsgUpdateChannelPermissions",
            value: exports.MsgUpdateChannelPermissions.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgUpdateChannelPermissions.decode(message.value);
    },
    toProto(message) {
        return exports.MsgUpdateChannelPermissions.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
            value: exports.MsgUpdateChannelPermissions.encode(message).finish()
        };
    }
};
function createBaseMsgUpdateChannelPermissionsResponse() {
    return {};
}
exports.MsgUpdateChannelPermissionsResponse = {
    typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissionsResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateChannelPermissionsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateChannelPermissionsResponse();
        return message;
    },
    fromSDK(_) {
        return {};
    },
    toSDK(_) {
        const obj = {};
        return obj;
    },
    fromAmino(_) {
        const message = createBaseMsgUpdateChannelPermissionsResponse();
        return message;
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgUpdateChannelPermissionsResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/MsgUpdateChannelPermissionsResponse",
            value: exports.MsgUpdateChannelPermissionsResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgUpdateChannelPermissionsResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgUpdateChannelPermissionsResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissionsResponse",
            value: exports.MsgUpdateChannelPermissionsResponse.encode(message).finish()
        };
    }
};
function createBaseMsgMintModuleTokens() {
    return {
        authority: "",
        amount: ""
    };
}
exports.MsgMintModuleTokens = {
    typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.amount !== "") {
            writer.uint32(18).string(message.amount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgMintModuleTokens();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.amount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            authority: (0, helpers_1.isSet)(object.authority) ? String(object.authority) : "",
            amount: (0, helpers_1.isSet)(object.amount) ? String(object.amount) : ""
        };
    },
    toJSON(message) {
        const obj = {};
        message.authority !== undefined && (obj.authority = message.authority);
        message.amount !== undefined && (obj.amount = message.amount);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseMsgMintModuleTokens();
        message.authority = object.authority ?? "";
        message.amount = object.amount ?? "";
        return message;
    },
    fromSDK(object) {
        return {
            authority: object?.authority,
            amount: object?.amount
        };
    },
    toSDK(message) {
        const obj = {};
        obj.authority = message.authority;
        obj.amount = message.amount;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseMsgMintModuleTokens();
        if (object.authority !== undefined && object.authority !== null) {
            message.authority = object.authority;
        }
        if (object.amount !== undefined && object.amount !== null) {
            message.amount = object.amount;
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.authority = message.authority;
        obj.amount = message.amount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgMintModuleTokens.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/MsgMintModuleTokens",
            value: exports.MsgMintModuleTokens.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgMintModuleTokens.decode(message.value);
    },
    toProto(message) {
        return exports.MsgMintModuleTokens.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
            value: exports.MsgMintModuleTokens.encode(message).finish()
        };
    }
};
function createBaseMsgMintModuleTokensResponse() {
    return {};
}
exports.MsgMintModuleTokensResponse = {
    typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokensResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgMintModuleTokensResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    fromPartial(_) {
        const message = createBaseMsgMintModuleTokensResponse();
        return message;
    },
    fromSDK(_) {
        return {};
    },
    toSDK(_) {
        const obj = {};
        return obj;
    },
    fromAmino(_) {
        const message = createBaseMsgMintModuleTokensResponse();
        return message;
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgMintModuleTokensResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "cosmos-sdk/MsgMintModuleTokensResponse",
            value: exports.MsgMintModuleTokensResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgMintModuleTokensResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgMintModuleTokensResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokensResponse",
            value: exports.MsgMintModuleTokensResponse.encode(message).finish()
        };
    }
};
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.UpdateParams = this.UpdateParams.bind(this);
        this.UpdateChannelPermissions = this.UpdateChannelPermissions.bind(this);
        this.MintModuleTokens = this.MintModuleTokens.bind(this);
    }
    UpdateParams(request) {
        const data = exports.MsgUpdateParams.encode(request).finish();
        const promise = this.rpc.request("cosmos.crosschain.v1.Msg", "UpdateParams", data);
        return promise.then(data => exports.MsgUpdateParamsResponse.decode(new _m0.Reader(data)));
    }
    UpdateChannelPermissions(request) {
        const data = exports.MsgUpdateChannelPermissions.encode(request).finish();
        const promise = this.rpc.request("cosmos.crosschain.v1.Msg", "UpdateChannelPermissions", data);
        return promise.then(data => exports.MsgUpdateChannelPermissionsResponse.decode(new _m0.Reader(data)));
    }
    MintModuleTokens(request) {
        const data = exports.MsgMintModuleTokens.encode(request).finish();
        const promise = this.rpc.request("cosmos.crosschain.v1.Msg", "MintModuleTokens", data);
        return promise.then(data => exports.MsgMintModuleTokensResponse.decode(new _m0.Reader(data)));
    }
}
exports.MsgClientImpl = MsgClientImpl;
