// @bnb-chain/greenfield-cosmos-types/cosmos/crosschain/v1/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/cosmos.crosschain.v1.MsgUpdateParams", tx_1.MsgUpdateParams], ["/cosmos.crosschain.v1.MsgUpdateChannelPermissions", tx_1.MsgUpdateChannelPermissions], ["/cosmos.crosschain.v1.MsgMintModuleTokens", tx_1.MsgMintModuleTokens]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        },
        updateChannelPermissions(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
                value: tx_1.MsgUpdateChannelPermissions.encode(value).finish()
            };
        },
        mintModuleTokens(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
                value: tx_1.MsgMintModuleTokens.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
                value
            };
        },
        updateChannelPermissions(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
                value
            };
        },
        mintModuleTokens(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
                value
            };
        }
    },
    toJSON: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        },
        updateChannelPermissions(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
                value: tx_1.MsgUpdateChannelPermissions.toJSON(value)
            };
        },
        mintModuleTokens(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
                value: tx_1.MsgMintModuleTokens.toJSON(value)
            };
        }
    },
    fromJSON: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        },
        updateChannelPermissions(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
                value: tx_1.MsgUpdateChannelPermissions.fromJSON(value)
            };
        },
        mintModuleTokens(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
                value: tx_1.MsgMintModuleTokens.fromJSON(value)
            };
        }
    },
    fromPartial: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        },
        updateChannelPermissions(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgUpdateChannelPermissions",
                value: tx_1.MsgUpdateChannelPermissions.fromPartial(value)
            };
        },
        mintModuleTokens(value) {
            return {
                typeUrl: "/cosmos.crosschain.v1.MsgMintModuleTokens",
                value: tx_1.MsgMintModuleTokens.fromPartial(value)
            };
        }
    }
};
