// @bnb-chain/greenfield-cosmos-types/greenfield/payment/events.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventFeePreview = exports.EventWithdraw = exports.EventDeposit = exports.EventForceSettle = exports.EventStreamRecordUpdate = exports.EventPaymentAccountUpdate = exports.feePreviewTypeToJSON = exports.feePreviewTypeFromJSON = exports.FeePreviewTypeAmino = exports.FeePreviewTypeSDKType = exports.FeePreviewType = exports.protobufPackage = void 0;
//@ts-nocheck
/* eslint-disable */
const stream_record_1 = require("./stream_record");
const helpers_1 = require("../../helpers");
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "greenfield.payment";
var FeePreviewType;
(function (FeePreviewType) {
    FeePreviewType[FeePreviewType["FEE_PREVIEW_TYPE_PRELOCKED_FEE"] = 0] = "FEE_PREVIEW_TYPE_PRELOCKED_FEE";
    FeePreviewType[FeePreviewType["FEE_PREVIEW_TYPE_UNLOCKED_FEE"] = 1] = "FEE_PREVIEW_TYPE_UNLOCKED_FEE";
    FeePreviewType[FeePreviewType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(FeePreviewType = exports.FeePreviewType || (exports.FeePreviewType = {}));
exports.FeePreviewTypeSDKType = FeePreviewType;
exports.FeePreviewTypeAmino = FeePreviewType;
function feePreviewTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "FEE_PREVIEW_TYPE_PRELOCKED_FEE":
            return FeePreviewType.FEE_PREVIEW_TYPE_PRELOCKED_FEE;
        case 1:
        case "FEE_PREVIEW_TYPE_UNLOCKED_FEE":
            return FeePreviewType.FEE_PREVIEW_TYPE_UNLOCKED_FEE;
        case -1:
        case "UNRECOGNIZED":
        default:
            return FeePreviewType.UNRECOGNIZED;
    }
}
exports.feePreviewTypeFromJSON = feePreviewTypeFromJSON;
function feePreviewTypeToJSON(object) {
    switch (object) {
        case FeePreviewType.FEE_PREVIEW_TYPE_PRELOCKED_FEE:
            return "FEE_PREVIEW_TYPE_PRELOCKED_FEE";
        case FeePreviewType.FEE_PREVIEW_TYPE_UNLOCKED_FEE:
            return "FEE_PREVIEW_TYPE_UNLOCKED_FEE";
        case FeePreviewType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.feePreviewTypeToJSON = feePreviewTypeToJSON;
function createBaseEventPaymentAccountUpdate() {
    return {
        addr: "",
        owner: "",
        refundable: false
    };
}
exports.EventPaymentAccountUpdate = {
    typeUrl: "/greenfield.payment.EventPaymentAccountUpdate",
    encode(message, writer = _m0.Writer.create()) {
        if (message.addr !== "") {
            writer.uint32(10).string(message.addr);
        }
        if (message.owner !== "") {
            writer.uint32(18).string(message.owner);
        }
        if (message.refundable === true) {
            writer.uint32(24).bool(message.refundable);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventPaymentAccountUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.addr = reader.string();
                    break;
                case 2:
                    message.owner = reader.string();
                    break;
                case 3:
                    message.refundable = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            addr: (0, helpers_1.isSet)(object.addr) ? String(object.addr) : "",
            owner: (0, helpers_1.isSet)(object.owner) ? String(object.owner) : "",
            refundable: (0, helpers_1.isSet)(object.refundable) ? Boolean(object.refundable) : false
        };
    },
    toJSON(message) {
        const obj = {};
        message.addr !== undefined && (obj.addr = message.addr);
        message.owner !== undefined && (obj.owner = message.owner);
        message.refundable !== undefined && (obj.refundable = message.refundable);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventPaymentAccountUpdate();
        message.addr = object.addr ?? "";
        message.owner = object.owner ?? "";
        message.refundable = object.refundable ?? false;
        return message;
    },
    fromSDK(object) {
        return {
            addr: object?.addr,
            owner: object?.owner,
            refundable: object?.refundable
        };
    },
    toSDK(message) {
        const obj = {};
        obj.addr = message.addr;
        obj.owner = message.owner;
        obj.refundable = message.refundable;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseEventPaymentAccountUpdate();
        if (object.addr !== undefined && object.addr !== null) {
            message.addr = object.addr;
        }
        if (object.owner !== undefined && object.owner !== null) {
            message.owner = object.owner;
        }
        if (object.refundable !== undefined && object.refundable !== null) {
            message.refundable = object.refundable;
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.addr = message.addr;
        obj.owner = message.owner;
        obj.refundable = message.refundable;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EventPaymentAccountUpdate.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return exports.EventPaymentAccountUpdate.decode(message.value);
    },
    toProto(message) {
        return exports.EventPaymentAccountUpdate.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/greenfield.payment.EventPaymentAccountUpdate",
            value: exports.EventPaymentAccountUpdate.encode(message).finish()
        };
    }
};
function createBaseEventStreamRecordUpdate() {
    return {
        account: "",
        crudTimestamp: helpers_1.Long.ZERO,
        netflowRate: "",
        frozenNetflowRate: "",
        staticBalance: "",
        bufferBalance: "",
        lockBalance: "",
        status: 0,
        settleTimestamp: helpers_1.Long.ZERO
    };
}
exports.EventStreamRecordUpdate = {
    typeUrl: "/greenfield.payment.EventStreamRecordUpdate",
    encode(message, writer = _m0.Writer.create()) {
        if (message.account !== "") {
            writer.uint32(10).string(message.account);
        }
        if (!message.crudTimestamp.isZero()) {
            writer.uint32(16).int64(message.crudTimestamp);
        }
        if (message.netflowRate !== "") {
            writer.uint32(26).string(message.netflowRate);
        }
        if (message.frozenNetflowRate !== "") {
            writer.uint32(34).string(message.frozenNetflowRate);
        }
        if (message.staticBalance !== "") {
            writer.uint32(42).string(message.staticBalance);
        }
        if (message.bufferBalance !== "") {
            writer.uint32(50).string(message.bufferBalance);
        }
        if (message.lockBalance !== "") {
            writer.uint32(58).string(message.lockBalance);
        }
        if (message.status !== 0) {
            writer.uint32(64).int32(message.status);
        }
        if (!message.settleTimestamp.isZero()) {
            writer.uint32(72).int64(message.settleTimestamp);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventStreamRecordUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.account = reader.string();
                    break;
                case 2:
                    message.crudTimestamp = reader.int64();
                    break;
                case 3:
                    message.netflowRate = reader.string();
                    break;
                case 4:
                    message.frozenNetflowRate = reader.string();
                    break;
                case 5:
                    message.staticBalance = reader.string();
                    break;
                case 6:
                    message.bufferBalance = reader.string();
                    break;
                case 7:
                    message.lockBalance = reader.string();
                    break;
                case 8:
                    message.status = reader.int32();
                    break;
                case 9:
                    message.settleTimestamp = reader.int64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            account: (0, helpers_1.isSet)(object.account) ? String(object.account) : "",
            crudTimestamp: (0, helpers_1.isSet)(object.crudTimestamp) ? helpers_1.Long.fromValue(object.crudTimestamp) : helpers_1.Long.ZERO,
            netflowRate: (0, helpers_1.isSet)(object.netflowRate) ? String(object.netflowRate) : "",
            frozenNetflowRate: (0, helpers_1.isSet)(object.frozenNetflowRate) ? String(object.frozenNetflowRate) : "",
            staticBalance: (0, helpers_1.isSet)(object.staticBalance) ? String(object.staticBalance) : "",
            bufferBalance: (0, helpers_1.isSet)(object.bufferBalance) ? String(object.bufferBalance) : "",
            lockBalance: (0, helpers_1.isSet)(object.lockBalance) ? String(object.lockBalance) : "",
            status: (0, helpers_1.isSet)(object.status) ? (0, stream_record_1.streamAccountStatusFromJSON)(object.status) : -1,
            settleTimestamp: (0, helpers_1.isSet)(object.settleTimestamp) ? helpers_1.Long.fromValue(object.settleTimestamp) : helpers_1.Long.ZERO
        };
    },
    toJSON(message) {
        const obj = {};
        message.account !== undefined && (obj.account = message.account);
        message.crudTimestamp !== undefined && (obj.crudTimestamp = (message.crudTimestamp || helpers_1.Long.ZERO).toString());
        message.netflowRate !== undefined && (obj.netflowRate = message.netflowRate);
        message.frozenNetflowRate !== undefined && (obj.frozenNetflowRate = message.frozenNetflowRate);
        message.staticBalance !== undefined && (obj.staticBalance = message.staticBalance);
        message.bufferBalance !== undefined && (obj.bufferBalance = message.bufferBalance);
        message.lockBalance !== undefined && (obj.lockBalance = message.lockBalance);
        message.status !== undefined && (obj.status = (0, stream_record_1.streamAccountStatusToJSON)(message.status));
        message.settleTimestamp !== undefined && (obj.settleTimestamp = (message.settleTimestamp || helpers_1.Long.ZERO).toString());
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventStreamRecordUpdate();
        message.account = object.account ?? "";
        message.crudTimestamp = object.crudTimestamp !== undefined && object.crudTimestamp !== null ? helpers_1.Long.fromValue(object.crudTimestamp) : helpers_1.Long.ZERO;
        message.netflowRate = object.netflowRate ?? "";
        message.frozenNetflowRate = object.frozenNetflowRate ?? "";
        message.staticBalance = object.staticBalance ?? "";
        message.bufferBalance = object.bufferBalance ?? "";
        message.lockBalance = object.lockBalance ?? "";
        message.status = object.status ?? 0;
        message.settleTimestamp = object.settleTimestamp !== undefined && object.settleTimestamp !== null ? helpers_1.Long.fromValue(object.settleTimestamp) : helpers_1.Long.ZERO;
        return message;
    },
    fromSDK(object) {
        return {
            account: object?.account,
            crudTimestamp: object?.crud_timestamp,
            netflowRate: object?.netflow_rate,
            frozenNetflowRate: object?.frozen_netflow_rate,
            staticBalance: object?.static_balance,
            bufferBalance: object?.buffer_balance,
            lockBalance: object?.lock_balance,
            status: (0, helpers_1.isSet)(object.status) ? (0, stream_record_1.streamAccountStatusFromJSON)(object.status) : -1,
            settleTimestamp: object?.settle_timestamp
        };
    },
    toSDK(message) {
        const obj = {};
        obj.account = message.account;
        obj.crud_timestamp = message.crudTimestamp;
        obj.netflow_rate = message.netflowRate;
        obj.frozen_netflow_rate = message.frozenNetflowRate;
        obj.static_balance = message.staticBalance;
        obj.buffer_balance = message.bufferBalance;
        obj.lock_balance = message.lockBalance;
        message.status !== undefined && (obj.status = (0, stream_record_1.streamAccountStatusToJSON)(message.status));
        obj.settle_timestamp = message.settleTimestamp;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseEventStreamRecordUpdate();
        if (object.account !== undefined && object.account !== null) {
            message.account = object.account;
        }
        if (object.crud_timestamp !== undefined && object.crud_timestamp !== null) {
            message.crudTimestamp = helpers_1.Long.fromString(object.crud_timestamp);
        }
        if (object.netflow_rate !== undefined && object.netflow_rate !== null) {
            message.netflowRate = object.netflow_rate;
        }
        if (object.frozen_netflow_rate !== undefined && object.frozen_netflow_rate !== null) {
            message.frozenNetflowRate = object.frozen_netflow_rate;
        }
        if (object.static_balance !== undefined && object.static_balance !== null) {
            message.staticBalance = object.static_balance;
        }
        if (object.buffer_balance !== undefined && object.buffer_balance !== null) {
            message.bufferBalance = object.buffer_balance;
        }
        if (object.lock_balance !== undefined && object.lock_balance !== null) {
            message.lockBalance = object.lock_balance;
        }
        if (object.status !== undefined && object.status !== null) {
            message.status = (0, stream_record_1.streamAccountStatusFromJSON)(object.status);
        }
        if (object.settle_timestamp !== undefined && object.settle_timestamp !== null) {
            message.settleTimestamp = helpers_1.Long.fromString(object.settle_timestamp);
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.account = message.account;
        obj.crud_timestamp = message.crudTimestamp ? message.crudTimestamp.toString() : undefined;
        obj.netflow_rate = message.netflowRate;
        obj.frozen_netflow_rate = message.frozenNetflowRate;
        obj.static_balance = message.staticBalance;
        obj.buffer_balance = message.bufferBalance;
        obj.lock_balance = message.lockBalance;
        obj.status = (0, stream_record_1.streamAccountStatusToJSON)(message.status);
        obj.settle_timestamp = message.settleTimestamp ? message.settleTimestamp.toString() : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EventStreamRecordUpdate.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return exports.EventStreamRecordUpdate.decode(message.value);
    },
    toProto(message) {
        return exports.EventStreamRecordUpdate.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/greenfield.payment.EventStreamRecordUpdate",
            value: exports.EventStreamRecordUpdate.encode(message).finish()
        };
    }
};
function createBaseEventForceSettle() {
    return {
        addr: "",
        settledBalance: ""
    };
}
exports.EventForceSettle = {
    typeUrl: "/greenfield.payment.EventForceSettle",
    encode(message, writer = _m0.Writer.create()) {
        if (message.addr !== "") {
            writer.uint32(10).string(message.addr);
        }
        if (message.settledBalance !== "") {
            writer.uint32(18).string(message.settledBalance);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventForceSettle();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.addr = reader.string();
                    break;
                case 2:
                    message.settledBalance = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            addr: (0, helpers_1.isSet)(object.addr) ? String(object.addr) : "",
            settledBalance: (0, helpers_1.isSet)(object.settledBalance) ? String(object.settledBalance) : ""
        };
    },
    toJSON(message) {
        const obj = {};
        message.addr !== undefined && (obj.addr = message.addr);
        message.settledBalance !== undefined && (obj.settledBalance = message.settledBalance);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventForceSettle();
        message.addr = object.addr ?? "";
        message.settledBalance = object.settledBalance ?? "";
        return message;
    },
    fromSDK(object) {
        return {
            addr: object?.addr,
            settledBalance: object?.settled_balance
        };
    },
    toSDK(message) {
        const obj = {};
        obj.addr = message.addr;
        obj.settled_balance = message.settledBalance;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseEventForceSettle();
        if (object.addr !== undefined && object.addr !== null) {
            message.addr = object.addr;
        }
        if (object.settled_balance !== undefined && object.settled_balance !== null) {
            message.settledBalance = object.settled_balance;
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.addr = message.addr;
        obj.settled_balance = message.settledBalance;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EventForceSettle.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return exports.EventForceSettle.decode(message.value);
    },
    toProto(message) {
        return exports.EventForceSettle.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/greenfield.payment.EventForceSettle",
            value: exports.EventForceSettle.encode(message).finish()
        };
    }
};
function createBaseEventDeposit() {
    return {
        from: "",
        to: "",
        amount: ""
    };
}
exports.EventDeposit = {
    typeUrl: "/greenfield.payment.EventDeposit",
    encode(message, writer = _m0.Writer.create()) {
        if (message.from !== "") {
            writer.uint32(10).string(message.from);
        }
        if (message.to !== "") {
            writer.uint32(18).string(message.to);
        }
        if (message.amount !== "") {
            writer.uint32(26).string(message.amount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventDeposit();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.from = reader.string();
                    break;
                case 2:
                    message.to = reader.string();
                    break;
                case 3:
                    message.amount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            from: (0, helpers_1.isSet)(object.from) ? String(object.from) : "",
            to: (0, helpers_1.isSet)(object.to) ? String(object.to) : "",
            amount: (0, helpers_1.isSet)(object.amount) ? String(object.amount) : ""
        };
    },
    toJSON(message) {
        const obj = {};
        message.from !== undefined && (obj.from = message.from);
        message.to !== undefined && (obj.to = message.to);
        message.amount !== undefined && (obj.amount = message.amount);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventDeposit();
        message.from = object.from ?? "";
        message.to = object.to ?? "";
        message.amount = object.amount ?? "";
        return message;
    },
    fromSDK(object) {
        return {
            from: object?.from,
            to: object?.to,
            amount: object?.amount
        };
    },
    toSDK(message) {
        const obj = {};
        obj.from = message.from;
        obj.to = message.to;
        obj.amount = message.amount;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseEventDeposit();
        if (object.from !== undefined && object.from !== null) {
            message.from = object.from;
        }
        if (object.to !== undefined && object.to !== null) {
            message.to = object.to;
        }
        if (object.amount !== undefined && object.amount !== null) {
            message.amount = object.amount;
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.from = message.from;
        obj.to = message.to;
        obj.amount = message.amount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EventDeposit.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return exports.EventDeposit.decode(message.value);
    },
    toProto(message) {
        return exports.EventDeposit.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/greenfield.payment.EventDeposit",
            value: exports.EventDeposit.encode(message).finish()
        };
    }
};
function createBaseEventWithdraw() {
    return {
        to: "",
        from: "",
        amount: ""
    };
}
exports.EventWithdraw = {
    typeUrl: "/greenfield.payment.EventWithdraw",
    encode(message, writer = _m0.Writer.create()) {
        if (message.to !== "") {
            writer.uint32(10).string(message.to);
        }
        if (message.from !== "") {
            writer.uint32(18).string(message.from);
        }
        if (message.amount !== "") {
            writer.uint32(26).string(message.amount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventWithdraw();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.to = reader.string();
                    break;
                case 2:
                    message.from = reader.string();
                    break;
                case 3:
                    message.amount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            to: (0, helpers_1.isSet)(object.to) ? String(object.to) : "",
            from: (0, helpers_1.isSet)(object.from) ? String(object.from) : "",
            amount: (0, helpers_1.isSet)(object.amount) ? String(object.amount) : ""
        };
    },
    toJSON(message) {
        const obj = {};
        message.to !== undefined && (obj.to = message.to);
        message.from !== undefined && (obj.from = message.from);
        message.amount !== undefined && (obj.amount = message.amount);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventWithdraw();
        message.to = object.to ?? "";
        message.from = object.from ?? "";
        message.amount = object.amount ?? "";
        return message;
    },
    fromSDK(object) {
        return {
            to: object?.to,
            from: object?.from,
            amount: object?.amount
        };
    },
    toSDK(message) {
        const obj = {};
        obj.to = message.to;
        obj.from = message.from;
        obj.amount = message.amount;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseEventWithdraw();
        if (object.to !== undefined && object.to !== null) {
            message.to = object.to;
        }
        if (object.from !== undefined && object.from !== null) {
            message.from = object.from;
        }
        if (object.amount !== undefined && object.amount !== null) {
            message.amount = object.amount;
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.to = message.to;
        obj.from = message.from;
        obj.amount = message.amount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EventWithdraw.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return exports.EventWithdraw.decode(message.value);
    },
    toProto(message) {
        return exports.EventWithdraw.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/greenfield.payment.EventWithdraw",
            value: exports.EventWithdraw.encode(message).finish()
        };
    }
};
function createBaseEventFeePreview() {
    return {
        account: "",
        feePreviewType: 0,
        amount: ""
    };
}
exports.EventFeePreview = {
    typeUrl: "/greenfield.payment.EventFeePreview",
    encode(message, writer = _m0.Writer.create()) {
        if (message.account !== "") {
            writer.uint32(10).string(message.account);
        }
        if (message.feePreviewType !== 0) {
            writer.uint32(16).int32(message.feePreviewType);
        }
        if (message.amount !== "") {
            writer.uint32(26).string(message.amount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventFeePreview();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.account = reader.string();
                    break;
                case 2:
                    message.feePreviewType = reader.int32();
                    break;
                case 3:
                    message.amount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            account: (0, helpers_1.isSet)(object.account) ? String(object.account) : "",
            feePreviewType: (0, helpers_1.isSet)(object.feePreviewType) ? feePreviewTypeFromJSON(object.feePreviewType) : -1,
            amount: (0, helpers_1.isSet)(object.amount) ? String(object.amount) : ""
        };
    },
    toJSON(message) {
        const obj = {};
        message.account !== undefined && (obj.account = message.account);
        message.feePreviewType !== undefined && (obj.feePreviewType = feePreviewTypeToJSON(message.feePreviewType));
        message.amount !== undefined && (obj.amount = message.amount);
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventFeePreview();
        message.account = object.account ?? "";
        message.feePreviewType = object.feePreviewType ?? 0;
        message.amount = object.amount ?? "";
        return message;
    },
    fromSDK(object) {
        return {
            account: object?.account,
            feePreviewType: (0, helpers_1.isSet)(object.fee_preview_type) ? feePreviewTypeFromJSON(object.fee_preview_type) : -1,
            amount: object?.amount
        };
    },
    toSDK(message) {
        const obj = {};
        obj.account = message.account;
        message.feePreviewType !== undefined && (obj.fee_preview_type = feePreviewTypeToJSON(message.feePreviewType));
        obj.amount = message.amount;
        return obj;
    },
    fromAmino(object) {
        const message = createBaseEventFeePreview();
        if (object.account !== undefined && object.account !== null) {
            message.account = object.account;
        }
        if (object.fee_preview_type !== undefined && object.fee_preview_type !== null) {
            message.feePreviewType = feePreviewTypeFromJSON(object.fee_preview_type);
        }
        if (object.amount !== undefined && object.amount !== null) {
            message.amount = object.amount;
        }
        return message;
    },
    toAmino(message) {
        const obj = {};
        obj.account = message.account;
        obj.fee_preview_type = feePreviewTypeToJSON(message.feePreviewType);
        obj.amount = message.amount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EventFeePreview.fromAmino(object.value);
    },
    fromProtoMsg(message) {
        return exports.EventFeePreview.decode(message.value);
    },
    toProto(message) {
        return exports.EventFeePreview.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/greenfield.payment.EventFeePreview",
            value: exports.EventFeePreview.encode(message).finish()
        };
    }
};
