// @bnb-chain/greenfield-cosmos-types/cosmos/gashub/v1beta1/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/cosmos.gashub.v1beta1.MsgUpdateParams", tx_1.MsgUpdateParams], ["/cosmos.gashub.v1beta1.MsgSetMsgGasParams", tx_1.MsgSetMsgGasParams]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        },
        setMsgGasParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgSetMsgGasParams",
                value: tx_1.MsgSetMsgGasParams.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgUpdateParams",
                value
            };
        },
        setMsgGasParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgSetMsgGasParams",
                value
            };
        }
    },
    toJSON: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        },
        setMsgGasParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgSetMsgGasParams",
                value: tx_1.MsgSetMsgGasParams.toJSON(value)
            };
        }
    },
    fromJSON: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        },
        setMsgGasParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgSetMsgGasParams",
                value: tx_1.MsgSetMsgGasParams.fromJSON(value)
            };
        }
    },
    fromPartial: {
        updateParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        },
        setMsgGasParams(value) {
            return {
                typeUrl: "/cosmos.gashub.v1beta1.MsgSetMsgGasParams",
                value: tx_1.MsgSetMsgGasParams.fromPartial(value)
            };
        }
    }
};
