// @bnb-chain/greenfield-cosmos-types/greenfield/challenge/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/greenfield.challenge.MsgSubmit", tx_1.MsgSubmit], ["/greenfield.challenge.MsgAttest", tx_1.MsgAttest], ["/greenfield.challenge.MsgUpdateParams", tx_1.MsgUpdateParams]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        submit(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgSubmit",
                value: tx_1.MsgSubmit.encode(value).finish()
            };
        },
        attest(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgAttest",
                value: tx_1.MsgAttest.encode(value).finish()
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        submit(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgSubmit",
                value
            };
        },
        attest(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgAttest",
                value
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgUpdateParams",
                value
            };
        }
    },
    toJSON: {
        submit(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgSubmit",
                value: tx_1.MsgSubmit.toJSON(value)
            };
        },
        attest(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgAttest",
                value: tx_1.MsgAttest.toJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        }
    },
    fromJSON: {
        submit(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgSubmit",
                value: tx_1.MsgSubmit.fromJSON(value)
            };
        },
        attest(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgAttest",
                value: tx_1.MsgAttest.fromJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        }
    },
    fromPartial: {
        submit(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgSubmit",
                value: tx_1.MsgSubmit.fromPartial(value)
            };
        },
        attest(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgAttest",
                value: tx_1.MsgAttest.fromPartial(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.challenge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        }
    }
};
