// @bnb-chain/greenfield-cosmos-types/greenfield/bridge/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/greenfield.bridge.MsgTransferOut", tx_1.MsgTransferOut], ["/greenfield.bridge.MsgUpdateParams", tx_1.MsgUpdateParams]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        transferOut(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgTransferOut",
                value: tx_1.MsgTransferOut.encode(value).finish()
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        transferOut(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgTransferOut",
                value
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgUpdateParams",
                value
            };
        }
    },
    toJSON: {
        transferOut(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgTransferOut",
                value: tx_1.MsgTransferOut.toJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.toJSON(value)
            };
        }
    },
    fromJSON: {
        transferOut(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgTransferOut",
                value: tx_1.MsgTransferOut.fromJSON(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromJSON(value)
            };
        }
    },
    fromPartial: {
        transferOut(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgTransferOut",
                value: tx_1.MsgTransferOut.fromPartial(value)
            };
        },
        updateParams(value) {
            return {
                typeUrl: "/greenfield.bridge.MsgUpdateParams",
                value: tx_1.MsgUpdateParams.fromPartial(value)
            };
        }
    }
};
